spark/src/main/java/org/opensearch/sql/spark/functions/response/DefaultSparkSqlFunctionResponseHandle.java

```diff
@@ -64,30 +67,43 @@ private static LinkedHashMap<String, ExprValue> extractRow(
-      } else if (type == ExprCoreType.SHORT) {
-        linkedHashMap.put(column.getName(), new ExprShortValue(row.getInt(column.getName())));
-      } else if (type == ExprCoreType.BYTE) {
-        linkedHashMap.put(column.getName(), new ExprByteValue(row.getInt(column.getName())));
-      } else if (type == ExprCoreType.DOUBLE) {
-        linkedHashMap.put(column.getName(), new ExprDoubleValue(row.getDouble(column.getName())));
-      } else if (type == ExprCoreType.FLOAT) {
-        linkedHashMap.put(column.getName(), new ExprFloatValue(row.getFloat(column.getName())));
-      } else if (type == ExprCoreType.DATE) {
-        // TODO :: correct this to ExprTimestampValue
-        linkedHashMap.put(column.getName(), new ExprStringValue(row.getString(column.getName())));
-      } else if (type == ExprCoreType.TIMESTAMP) {
-        linkedHashMap.put(
-            column.getName(), new ExprTimestampValue(row.getString(column.getName())));
-      } else if (type == ExprCoreType.STRING) {
-        linkedHashMap.put(column.getName(), new ExprStringValue(jsonString(row, column.getName())));
+      if (!row.has(column.getName())) {
+        linkedHashMap.put(column.getName(), ExprNullValue.of());
       } else {
-        throw new RuntimeException("Result contains invalid data type");
+        if (type == ExprCoreType.BOOLEAN) {
+          linkedHashMap.put(
+              column.getName(), ExprBooleanValue.of(row.getBoolean(column.getName())));
+        } else if (type == ExprCoreType.LONG) {
+          linkedHashMap.put(column.getName(), new ExprLongValue(row.getLong(column.getName())));
+        } else if (type == ExprCoreType.INTEGER) {
+          linkedHashMap.put(column.getName(), new ExprIntegerValue(row.getInt(column.getName())));
+        } else if (type == ExprCoreType.SHORT) {
+          linkedHashMap.put(column.getName(), new ExprShortValue(row.getInt(column.getName())));
+        } else if (type == ExprCoreType.BYTE) {
+          linkedHashMap.put(column.getName(), new ExprByteValue(row.getInt(column.getName())));
+        } else if (type == ExprCoreType.DOUBLE) {
+          linkedHashMap.put(column.getName(), new ExprDoubleValue(row.getDouble(column.getName())));
+        } else if (type == ExprCoreType.FLOAT) {
+          linkedHashMap.put(column.getName(), new ExprFloatValue(row.getFloat(column.getName())));
+        } else if (type == ExprCoreType.DATE) {
+          // TODO :: correct this to ExprTimestampValue
+          linkedHashMap.put(column.getName(), new ExprStringValue(row.getString(column.getName())));
+        } else if (type == ExprCoreType.TIMESTAMP) {
+          linkedHashMap.put(
+              column.getName(), new ExprTimestampValue(row.getString(column.getName())));
+        } else if (type == ExprCoreType.STRING) {
+          linkedHashMap.put(column.getName(), new ExprStringValue(row.getString(column.getName())));
+        } else {
```

Comment:
Just for understanding: anything apart from above datatypes is a SparkDataType.

Reply:
yes, any spark datatype sql can not mapping to is wrapped as SparkDataType.

The major reason for doing this mapping is event analysis frontend feature use SQL data type to display type icon.